File: inspect_excel.js
const xlsx = require('xlsx');

try {
  const workbook = xlsx.readFile('Agent_Pricing_Calculator_v3_Final - Copy.xlsx');
  console.log("Sheet names:", workbook.SheetNames);
  
  workbook.SheetNames.forEach(sheetName => {
    console.log(`\n--- Sheet: ${sheetName} ---`);
    const worksheet = workbook.Sheets[sheetName];
    const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });
    console.log(`Total Rows: ${data.length}`);
    
    // Print first 30 rows
    const preview = data.slice(0, 30);
    preview.forEach((row, i) => {
        console.log(`Row ${i + 1}: ${JSON.stringify(row)}`);
    });
  });
} catch (error) {
  console.error("Error reading file:", error.message);
}

File: assets/inspect_excel.py
import pandas as pd
import json

file_path = "Agent_Pricing_Calculator_v3_Final - Copy.xlsx"

try:
    xl = pd.ExcelFile(file_path)
    print("Sheet names:", xl.sheet_names)
    
    for sheet in xl.sheet_names:
        df = xl.parse(sheet)
        print(f"\n--- Sheet: {sheet} ---")
        print("Shape:", df.shape)
        # Print first 30 rows and all columns, replacing NaNs with empty strings for readability
        print(df.head(30).fillna('').to_string())
except Exception as e:
    print(f"Error reading file: {e}")
